modo/cram.py

```diff
@@ -10,20 +10,66 @@
 from rdflib import Graph
 import modo_schema.datamodel as model
 
+import os, sys, re
+import htsget
 from .helpers import parse_region
 
 
 def slice_cram(path: str, region: str) -> Iterator[AlignedSegment]:
     """Return an iterable slice of the CRAM file."""
 
-    chrom, start, stop = parse_region(region)
+    chrom, start, end = parse_region(region)
+    if start == "":
+        start = None
+    else:
+        start = int(start)
+    if end == "":
+        end = None
+    else:
+        end = int(end)
     cramfile = AlignmentFile(path, "rc")
 
-    iter = cramfile.fetch(chrom, start, stop)
+    iter = cramfile.fetch(chrom, start, end)
 
     return iter
 
 
+def slice_remote_cram(
+    url: str, region: str = None, output_filename: str = None
```

Comment:
__To-do__: `parse_region()` expects a str, but region is assigned  `None ` as default, which will fail. 
So I think `region` should either be a non-optional argument and we expect the user to specify a string or we use something like "*" as default.

Reply:
fixed - changed to 'region: Optional[str] = None'